### PR TITLE
Allow clicking the site editor frame to enter edit mode

### PR DESCRIPTION
**Pull request by youknowriad**
Related to #36667 

## What?

This PR simplifies the flows in the site editor by allowing users to click the "frame" to enter edit mode.

## Testing Instructions

1- Open the site editor.
2- you should be able to tab and click the frame or just click using the mouse to enter edit mode.

### CODE REVIEW

File: packages/edit-site/src/components/block-editor/style.scss [hunk 1]
```diff
@@ -158,3 +158,7 @@
 		box-shadow: inset 0 0 0 2px var(--wp-admin-theme-color);
 	}
 }
+
+.edit-site-visual-editor__editor-canvas.is-canvas-view {
+	cursor: pointer;
```
Comment (youknowriad) [inline packages/edit-site/src/components/block-editor/style.scss]: For some reason, I don't see the cursor changing, I'm not sure I understand the reason. @jasmussen any idea?

**Comment by jasmussen** [comment]
Nice, this is working well.

test

Eventually, but not for this PR as I think it needs a little design, I think we want a tooltip that says "Click to edit Home template", something like that. 

But we should use the `pointer` cursor when you're hovering the preview.

**Review by jasmussen — APPROVED** [review]
Designwise this feels like the right direction.

**Comment by youknowriad** [comment]
> But we should use the pointer cursor when you're hovering the preview.

Yeah, this is what I tried with my style above but it's not working, I'm not sure exactly why.

**Comment by carolinan** [comment]
Not a code review: I think this is an improvement. It feels intuitive to click on the canvas.
I wonder if the Edit button now needs to be as prominent?

**Comment by youknowriad** [comment]
Looks like the iframe element don't support "cursor". You need to apply the style within the iframe which is harder :).
Anyway, I'll find a way to solve that.

@carolinan After speaking with Joen, I think they have some design tweaks in the works that go into this direction.

**Comment by jasmussen** [comment]
Yeah the feedback around the edit button has been sufficient enough that it might be worth reconsidering it. 

One idea is to show an edit button next to each template, which could make much more sense. I hope one from the group can share a mockup to that effect soon!

**Comment by youknowriad** [comment]
Turns out adding the cursor is not an easy task here and have some big implications, I'm going to leave this separate for now. The current PR will do.

**Comment by jasmussen** [comment]
> Turns out adding the cursor is not an easy task here and have some big implications, I'm going to leave this separate for now. The current PR will do.

That's fine — as noted, I believe we will want to add edit buttons in context of each page or template instead of having a singular edit button at the top. In that vein, clicking the big site preview is more of an added secondary affordance.

**Comment by jameskoster** [comment]
We can figure it out alongside the cursor, but a hover effect might be nice. 

One other thought, now that the frame acts as one big button, does the scrollability seem a bit unexpected?

**Comment by jasmussen** [comment]
> One other thought, now that the frame acts as one big button, does the scrollability seem a bit unexpected?

I think it's valuable to have, and it could even be an argumet for keeping the default cursor. A snackbar-esque tooltip could inform of the ability to click or even double click. Main point being, scrolling it is useful for the preview aspect, and clicking to edit can hopefully become a secondary thing next to explicit edit buttons in context of pages and templates.

**Comment by Mamaduka** [comment]
I just cherry-picked this PR to the release/15.1 branch to get it included in the next release: 228c2de227

**Comment by paaljoachim** [comment]
Great! Thank you! This will be helpful!

**Pull request merged**